[license] Move `x-license-pro` cli in a new package

It used to be possible to generate a license key with

```
npx @mui/x-license-pro@4.0.1 --order 121
```

It's no longer possible, it's broken when you run this same command with the latest version. It outputs

> npx @mui/x-license-pro --order 121
Error: Cannot find module '../build/node/cli/license-cli'

It has been broken for a long time, and nobody complained about it. So, I 👍  the point that @m4theushw has, we should get ride of the whole bin and cli folders:

- https://unpkg.com/browse/@mui/x-license-pro@5.6.1/bin/
- https://unpkg.com/browse/@mui/x-license-pro@5.6.1/cli/

It would also mean we can get rid of 

https://github.com/mui/mui-x/blob/0c815ce1e9659c8f01403ef5fc6812251841a995/packages/x-license-pro/package.json#L38-L39

Which are polluting the dependencies:

<img width="654" alt="Screenshot 2022-03-21 at 17 36 42" src="https://user-images.githubusercontent.com/3165635/159310720-3859ce57-88a8-48ff-9637-b7209667c41f.png">

https://npm.anvaka.com/#/view/2d/%2540mui%252Fx-data-grid-pro

---

For though that wonder. How do customers get the license key? Will we break it? No, this is done with a WooCommerce webhook in https://github.com/mui/mui-store/blob/d0d86fc9e13d3cf0ca55eba9612ec00d1543356c/lambda/keymailer/index.js#L5

--- a/packages/grid/x-data-grid-generator/package.json
+++ b/packages/grid/x-data-grid-generator/package.json
@@ -39,9 +39,7 @@
   },
   "devDependencies": {
     "@types/chance": "^1.1.3",
-    "@types/lru-cache": "^7.10.9",
-    "esm": "^3.2.25",
-    "yargs": "^17.5.1"
+    "@types/lru-cache": "^7.10.9"
   },
   "peerDependencies": {
     "@mui/icons-material": "^5.4.1",

--- a/packages/x-license-pro/package.json
+++ b/packages/x-license-pro/package.json
@@ -31,9 +31,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.18.9",
-    "@mui/utils": "^5.9.3",
-    "esm": "^3.2.25",
-    "yargs": "^17.5.1"
+    "@mui/utils": "^5.9.3"
   },
   "peerDependencies": {
     "react": "^17.0.2 || ^18.0.0"

